server/index.js

```diff
@@ -0,0 +1,38 @@
+import express from 'express';
+import cors from 'cors';
+import sqlite3 from 'sqlite3'; // Import sqlite3
+import blogControllerRoutes from './Controllers/blogControllers.js';
+
+const app = express();
+const PORT = 3030;
+
+// Middleware
+app.use(cors());
+app.use(express.json());
+
+// Initialize SQLite3 database
+const db = new sqlite3.Database('deurdu.db', (err) => {
+  if (err) {
+    console.error('Error connecting to the database:', err);
+  } else {
+    console.log('Connected to the SQLite database');
+  }
+});
+
+// Make the database connection available to routes
+app.use((req, res, next) => {
+  req.db = db; // Attach the database connection to the request object
+  next();
```

Comment:
what does this `next` achieve?

Reply:
The `next()` function in Express middleware passes control to the next middleware function in the `request-response cycle`. Without calling next(), the request would hang, and the server wouldn't proceed to handle the next middleware or route.

```bash 
app.use((req, res, next) => {
  req.db = db;  // Attach database to request object
  next();       // Pass control to the next middleware
});

```